src/fiatconnect/saga.ts

```diff
@@ -69,17 +80,116 @@ export function* handleFetchFiatConnectQuotes({
       flow,
       fiatConnectCashInEnabled,
       fiatConnectCashOutEnabled,
-      fiatConnectProviders,
+      fiatConnectProviders: providerIds
+        ? fiatConnectProviders.filter(({ id }) => providerIds.includes(id))
+        : fiatConnectProviders,
     })
     yield put(fetchFiatConnectQuotesCompleted({ quotes }))
   } catch (error) {
     Logger.error(TAG, 'Could not fetch fiatconnect quotes', error)
     yield put(fetchFiatConnectQuotesFailed({ error: 'Could not fetch fiatconnect quotes' }))
   }
 }
+/**
+ * This saga attempts to fetch a quote and matching fiatAccount for a provider that the user
+ * has previously used in the CICO flow.
+ * If successful: Navigates the user to the FiatConnectReview screen
+ * If fail: Navigates the user to the SelectProvider screen
+ */
+export function* handleAttemptReturnUserFlow({
+  payload: params,
+}: ReturnType<typeof attemptReturnUserFlow>) {
+  const { amount, flow, selectedCrypto, providerId, fiatAccountId, fiatAccountType } = params
+  const digitalAsset = {
+    [Currency.Celo]: CiCoCurrency.CELO,
+    [Currency.Dollar]: CiCoCurrency.CUSD,
+    [Currency.Euro]: CiCoCurrency.CEUR,
+  }[selectedCrypto]
+  const fiatConnectProviders: FiatConnectProviderInfo[] | null = yield select(
+    fiatConnectProvidersSelector
+  )
+  try {
+    // Fetch Quote associated with the cached providerId
+    yield put(
+      fetchFiatConnectQuotes({
+        digitalAsset,
+        cryptoAmount: amount.crypto,
+        flow,
+        providerIds: [providerId],
+      })
+    )
+    const quotes: (FiatConnectQuoteSuccess | FiatConnectQuoteError)[] = yield select(
+      fiatConnectQuotesSelector
+    )
+    const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
+    const normalizedQuote = normalizedQuotes.find(
+      (q) => q.getFiatAccountType() === fiatAccountType && q.getProviderId() === providerId
+    )
+    if (!normalizedQuote) {
+      throw new Error('Could not find quote')
+    }
+    // Get the provider info
+    const fiatConnectProvider = fiatConnectProviders?.find((provider) => provider.id === providerId)
+    if (!fiatConnectProvider) {
+      throw new Error('Could not find provider')
+    }
+    // Fetch Fiat Account associated with the cached providerId / fiatAccountId
+    const fiatAccounts: FiatAccount[] = yield call(
```

Comment:
should this be invoked in parallel? Also we'd also want this if we don't have a provider stored or if a user has used two providers? We would need to do this in the select provider screen after provider selection before fiat details screen right? E.g.,
- cash out first time with Provider A
- cash out shows A, but user selects Provider B 
- cash out shows B, but user selects A --- this should go to the Review screen directly? since there's a stored fiat account?

Reply:
> should this be invoked in parallel

It can be yes! I can add that in.

Reply:
For the second part of your comment I'm a bit confused by it, but I was thinking for the SelectProviders screen that I'd fetch fiatAccounts for every quote that came back and to not rely on the cache for this part. 

Fetching featAccounts before click on the SelectProvider screen has the advantage of us being able to label which ones have already been linked on the SelectProvider screen which has been talked about in a few designs but I think was missed in the latest round.

Reply:
yeah I was just asking about the case when a user has used multiple providers. In that case we have to make the call to get accounts before showing the fiat details screen right? I am good If it's planned to be in a follow up PR